source/API_Reference/Event_Webhook/event.md

```diff
@@ -0,0 +1,1043 @@
+
+String categories:
+
+{% codeblock lang:json %}
+[
+  {
+    "email": "john.doe@sendgrid.com",
+    "timestamp": 1337966815,
+    "category": "newuser",
+    "event": "open"
+  },
+  {
+    "email": "jane.doe@sendgrid.com",
+    "timestamp": 1337966815,
+    "category": "olduser",
+    "event": "open"
+  }
+]
+{% endcodeblock %}
+
+Array:
+
+{% codeblock lang:json %}
+[
+  {
+    "email": "john.doe@sendgrid.com",
+    "timestamp": 1337966815,
+    "category": [
+      "newuser",
+      "transactional"
+    ],
+    "event": "open"
+  },
+  {
+    "email": "jane.doe@sendgrid.com",
+    "timestamp": 1337966815,
+    "category": "olduser",
+    "event": "open"
+  }
+]
+{% endcodeblock %}
+
+- <a name="asmgroupid"></a>`asm_group_id` - The ID of the unsubscribe group the recipients email is included in. ASM IDs correspond to the ID that is returned when you create an unsubscribe group.
+- <a name="uniqueargs"></a>`unique_args` or `custom_args` - 
+{% anchor h2 %}
+Unique Arguments and Custom Arguments
+{% endanchor %}
+
+Events generated by SendGrid may also include any [unique arguments]({{root_url}}/API_Reference/SMTP_API/unique_arguments.html) or custom arguments.
+
+{% info %}
+Unique arguments and custom arguments are functionally the same: It's called unique arguments in the SMTP API or v2 Mail Send, and it's called custom arguments in v3 Mail Send.
+{% endinfo %}
+
+{% anchor h3 %}
+Unique Arguments
+{% endanchor %}
+
+To define and receive unique arguments when sending email via the [SMTP API]({{root_url}}/API_Reference/SMTP_API/index.html) or the [v2 Mail Send endpoint]({{root_url}}/API_Reference/Web_API/mail.html) simply use the `unique_args` parameter in the X-SMTPAPI header or parameter. For example, if you have an application and want to receive custom parameters such as the `userid` and the email `template`, you would submit them with the X-SMTPAPI header or parameter, as described [here]({{root_url}}/API_Reference/SMTP_API/unique_arguments.html).
+
+For example, if you include the following unique arguments in your x-smtpapi parameter for an email sent via the v2 Mail Send endpoint:
+
+{% codeblock lang:json %}
+{
+  "unique_args": {
+    "userid": "1123",
+    "template": "welcome"
+  }
+}
+{% endcodeblock %}
+
+You will receive the same unique argument included with the data returned to the Event Webhook:
+
+{% codeblock lang:json %}
+[
+  {
+    "sg_message_id":"sendgrid_internal_message_id",
+    "email": "john.doe@sendgrid.com",
+    "timestamp": 1337966815,
+    "event": "click",
+    "url": "https://sendgrid.com",
+    "userid": "1123",
+    "template": "welcome"
+  }
+]
+{% endcodeblock %}
+
+{% warning %}
+You can create unique arguments with the same keys as reserved words, such as "event" or "email". In this case, SendGrid will default to the reserved key and NOT your unique argument. An example of this is below.
```

Comment:
In this case, SendGrid defaults to the reserved key and NOT your unique argument.